jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/PathResource.java

```diff
@@ -142,8 +143,8 @@ public static boolean isSameName(Path pathA, Path pathB)
 
     PathResource(URI uri, boolean bypassAllowedSchemeCheck)
     {
-        // normalize to referenced location, Paths.get() doesn't like "/bar/../foo/text.txt" style references
-        // and will return a Path that will not be found with `Files.exists()` or `Files.isDirectory()`
+        // Normalize to referenced location, Paths.get() doesn't like "/bar/../foo/text.txt" style references
+        // and will return a Path that will not be found with `Files.exists()` or `Files.isDirectory()`.
         this(Paths.get(uri.normalize()), uri, bypassAllowedSchemeCheck);
```

Comment:
When using `Paths.get(URI)` the URI has to be sane, as path segments like `/../` are treated as valid segments to resolve on the filesystem of the path.
If we were using `Paths.get(String)` then those segments (`/../`) are treated as navigation segments and will work.

The behavior here is because we use URI vs String when creating the Path object.